app/types.py

```diff
@@ -1,11 +1,42 @@
+from dataclasses import asdict, dataclass
 from datetime import datetime
-from typing import Optional
+from typing import List, NewType, Optional
 
-from app.encryption import NotificationDictToSign
 from app.models import Job, NotificationType, Service
+from app.queue import QueueMessage
 
+SignedNotification = NewType("SignedNotification", str)
+SignedNotifications = NewType("SignedNotifications", List[SignedNotification])
 
-class VerifiedNotification(NotificationDictToSign):
+
+@dataclass
+class PendingNotification(QueueMessage):
+    # todo: remove duplicate keys
+    # todo: remove all NotRequired and decide if key should be there or not
+    id: str
+    template: str  # actually template_id
+    service_id: str
+    template_version: int
+    to: str  # recipient
+    personalisation: Optional[dict]
+    simulated: bool
+    api_key: str
+    key_type: str  # should be ApiKeyType but can't import that here
+    client_reference: Optional[str]
+    reply_to_text: Optional[str]
```

Comment:
The `reply_to_text` field can be set via the API by providing the `email_reply_to_id` JSON field in the POST request. So that should be present in the `PendingNotification | VerifiedNotification` type and passed down the stream to the database.